src/Features/SupportWireLoading/BrowserTest.php

```diff
@@ -378,6 +381,64 @@ public function render()
         ->assertSee('Foo')
         ;
     }
+
+    /** @test */
+    function wire_loading_works_when_function_called_by_event()
+    {
+        Livewire::visit(new class extends Component {
+            protected $listeners = ['foo' => 'onFoo'];
+
+            function onFoo() {
+                sleep(5);
+            }
+
+            function render()
+            {
+                return <<<'HTML'
+                <div>
+                    <button @click="$dispatch('foo')" dusk="button">Dispatch "foo"</button>
+
+                    <div wire:loading wire:target="onFoo" dusk="loadingIndicator">
+                        Processing...
+                    </div>
+                </div>
+                HTML;
+            }
+        })
+        ->assertDontSee('Processing...')
+        ->waitForLivewire()->click('@button')
+        ->assertSee('Processing...');
+    }
+
+    /** @test */
+    function wire_loading_works_when_function_called_by_click()
+    {
+        $component = Livewire::visit(new class extends Component {
+            public function foo()
+            {
+                sleep(2);
+            }
+
+            public function render() { 
+                return <<<'HTML'
+                    <div>
+                        <button wire:click="foo" dusk="fooButton">Reset</button>
+                        <div wire:loading wire:target="foo" dusk="loadingIndicator">
+                            Processing...
+                        </div>
+                    </div>
+                HTML;
+             }
+        });
+
+        $component->assertDontSee('Processing...')
+            ->press('@fooButton')
+            ->waitForText('Processing...')
+            ->assertSee('Processing...')
+            ->pause(3100)
+            ->assertDontSee('Processing...')
+            ;
+    }
```

Comment:
```suggestion
```

I think you can delete this because it's not related with dispatching event (and this test is already exist)

Reply:
hi @danie-ramdhani, sorry for the ask but how that i build the livewire to test my changes made locally on js\directives\wire-loading.js ? I'v searched how to do this on Contribution Guide but not find. Sorry, i'm begginer on open source world's